### ISSUE TITLE
Redundant linker flag

### ISSUE THREAD

**Issue by eberkund** [dcuddeback/gphoto-rs#1]
```
   Compiling pkg-config v0.3.9
   Compiling libc v0.2.30
   Compiling gphoto2-sys v0.1.1
warning: redundant linker flag specified for library `m`

error[E0517]: attribute should be applied to struct, enum or union
  --> /home/erik/.cargo/registry/src/github.com-1ecc6299db9ec823/gphoto2-sys-0.1.1/src/port.rs:17:1
   |
17 | #[repr(C)]
   | ^^^^^^^^^^ requires a struct, enum or union

error: aborting due to previous error

error: Could not compile `gphoto2-sys`.
```

**Comment by charles-d-burton** [comment]
I'm getting this error too.

**Comment by dcuddeback** [comment]
The actual error causing your build to fail is the `#[repr(C)]` attribute:

```
error[E0517]: attribute should be applied to struct, enum or union
   |
17 | #[repr(C)]
   | ^^^^^^^^^^ requires a struct, enum or union
```

The redundant linker flag is just a warning, and something I have no control over since that's generated by `pkg-config` on the upstream library:

```
$ pkg-config --libs libgphoto2
-lgphoto2 -lm -lgphoto2_port -lm
```

Anyways, the root cause of your compilation error is fixed in v0.1.2. Thanks for reporting this.

**Comment by eberkund** [comment]
v0.1.2 fixes the issues I was previously having
